### PR TITLE
Try to customize UD with product.json. Remove all not needed files.

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### What does this PR do?

1. Add custom product.json file.
2. Remove all not needed file anymore
3. Clean up pom.xml

### CODE REVIEW

File: assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  "configuration": {
+    "menu": {
+      "disabled": ["organizations"]
+    }, 
+    "features": {
+      "disabled": ["workspaceSharing","kubernetesNamespaceSelector"]
+    },
+    "prefetch": {
+      "cheCDN":"/api/cdn-support/paths"
```
Comment (ibuziuk) [inline assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: @davidfestal could you please comment which one should be used here?

Comment (ibuziuk) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: wondering if we still need https://github.com/redhat-developer/rh-che/tree/master/plugins/fabric8-cdn-support/src/main/java/com/redhat/che/cdn ?

File: assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "workspace" : "/docs/che-7/workspaces-overview/",
+    "factory" : "/docs/factories-getting-started.html",
+    "organization": "/docs/organizations.html",
+    "converting": "/docs/che-7/converting-a-che-6-workspace-to-a-che-7-devfile/",
+    "general": "/docs/che-7",
+    "faq": "https://github.com/redhat-developer/rh-che/blob/master/FAQ.adoc"
+  },
+  "configuration": {
+    "menu": {
+      "disabled": ["organizations"]
```
Comment (ibuziuk) [inline assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: @slemeur I would probably also consider disabling factories on Hosted Che

**Comment by ibuziuk** [comment]
@sleshchenko is it expected that "dashboard" menu redirects to "Get Started". Do we need "Dashboard" in this case at all?

File: assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json (hunk 2 above)
Comment (slemeur) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: +1

**Comment by akurinnoy** [comment]
@ibuziuk This is expected behavior. Redirection occurs only if no workspaces have been created, so there is nothing to show on the dashboard page.

File: assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json (hunk 1 above)
Comment (davidfestal) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: I assume we still need it yes. I'm not aware of this having been migrated to the upstream Che server codebase.
And the URL should be the same as well.

Comment (ibuziuk) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: @davidfestal do we need to pre-fetch some `resoures` also ?
https://github.com/eclipse/che/pull/15889

Comment (davidfestal) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: I don't think so

**Comment by ibuziuk** [comment]
> @ibuziuk This is expected behavior. Redirection occurs only if no workspaces have been created, so there is nothing to show on the dashboard page.

@slemeur @akurinnoy I have created an issue for considering the removal of the "Dashboard" menu item at all - https://github.com/eclipse/che/issues/16040

File: assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "converting": "/docs/che-7/converting-a-che-6-workspace-to-a-che-7-devfile/",
+    "general": "/docs/che-7",
+    "faq": "https://github.com/redhat-developer/rh-che/blob/master/FAQ.adoc"
+  },
+  "configuration": {
+    "menu": {
+      "disabled": ["organizations", "factories"]
+    }, 
+    "features": {
+      "disabled": ["workspaceSharing","kubernetesNamespaceSelector"]
```
Comment (sleshchenko) [inline assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]:
```suggestion
      "disabled": ["workspaceSharing", "kubernetesNamespaceSelector"]
```

**Comment by ibuziuk** [comment]
@vparfonov could you please rebase on top of the latest version of `upstream-check-7.9.0-SNAPSHOT` ?

**Comment by ibuziuk** [comment]
@vparfonov could you please rebase on master now

**Comment by ibuziuk** [comment]
[test]

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
@ibuziuk @vparfonov The test issue was fixed, now we're waiting for the `Create Workspace` button.

**Comment by ibuziuk** [comment]
Upstream issue - https://github.com/eclipse/che/issues/16129
Should be fixed in 7.9.1 (putting PR on hold until than)

**Comment by ibuziuk** [comment]
Will be merged as part of `7.9.1` update - https://github.com/eclipse/che/issues/16129

File: assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "favicon": "favicon.ico",
+  "loader": "loader.svg",
+  "websocketContext": "/api/websocket",
+  "helpPath": "https://www.eclipse.org/che/",
+  "helpTitle": "Community",
+  "supportEmail": "che-dev@eclipse.org",
+  "oauthDocs": "Configure OAuth in the che.properties file.",
+  "workspace": {
+    "priorityStacks": ["Java", "Java-MySQL", "Blank"],
+    "defaultStack": "java-mysql"
```
Comment (olexii4) [inline assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: The quick fix (just adds there): 
```
"creationLink": "#/create-workspace"
```

Comment (olexii4) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: It is a default value.  But, it won't be used until merge this [PR](https://github.com/eclipse/che/pull/16175)

Comment (ibuziuk) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: @olexii4 thanks for the pointer!
@vparfonov could you please apply the suggestion to unblock the PR?

Comment (vparfonov) [reply assembly/fabric8-ide-dashboard-war/src/assets/branding/product.json]: done

**Pull request merged**